ocaml/xapi/repository.ml

```diff
@@ -191,21 +192,29 @@ let get_host_updates_by_repository ~__context enabled host () =
     raise Api_errors.(Server_error (get_host_updates_failed, [ref]))
 
 let set_available_updates ~__context =
-  with_pool_repositories @@ fun () ->
   ignore (get_single_enabled_update_repository ~__context) ;
   let hosts = Db.Host.get_all ~__context in
-  let enabled = get_enabled_repositories ~__context in
   let funs =
-    List.map
-      (fun h -> get_host_updates_by_repository ~__context enabled h)
-      hosts
+    List.map (fun host ->
+        fun () ->
+          ( host, (http_get_host_updates_in_json ~__context ~host ~installed:true) )
```

Comment:
`get_host_updates_by_repository` has previously called `http_get_host_updates_in_json` with `installed:false`, but it is called with `installed:true` here now, why?
I'd assume that installed:true is used here so that the cache will be useful for the updates query below, but I'm not sure about what effects it has on this function...

Reply:
When it was called `http_get_host_updates_in_json`, it only served for `sync_updates` purpose. `sync_updates` doesn't need the extra data returned with call parameter `installed:true`.
Now `get_host_updates_by_repository` serves for both `sync_udpates` and the cache which will be used by HTTP endpoint `/updates` handler later.